src/main/java/org/deepsymmetry/beatlink/dbserver/ConnectionManager.java

```diff
@@ -244,7 +251,12 @@ public void deviceFound(final DeviceAnnouncement announcement) {
                 return;
             }
             logger.debug("Processing device found, number: {}, name: {}", announcement.getDeviceNumber(), announcement.getDeviceName());
-            new Thread(() -> requestPlayerDBServerPort(announcement)).start();
+            final Thread queryThread = new Thread(() -> requestPlayerDBServerPort(announcement));
```

Comment:
We do need a small bit more protection against the fact that we are going to see multiple devices with the same IP address in short succession. The simplest way I could think of to do that is to keep track of which addresses we are already querying for, and not starting a new thread if one is already working. The `putIfAbsent()` method of `ConcurrentHashMap` is a lightweight but atomic way to ensure that. If it returns a `null` value, the new thread was added to the map, so we should start it. Otherwise, we just discard the thread unused, because we know another is working on that address.
The running thread will remove itself as it ends, see balow.